src/Microsoft.Data.SqlClient/src/Microsoft/Data/StreamExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿#if NETFRAMEWORK
+
+namespace System.IO
+{
+    // Helpers to read/write Span/Memory<byte> to Stream before netstandard 2.1
+    internal static class StreamExtensions
+    {
+        internal static void ReadExactly(this Stream stream, byte[] buffer, int offset, int count)
+        {
+            var totalRead = 0;
+            do
+            {
+                var read = stream.Read(buffer, offset + totalRead, count - totalRead);
+                if (read is 0)
+                    throw new EndOfStreamException();
```

Comment:
Stop it! I don't want to see if statements without braces in new code!

https://www.imperialviolet.org/2014/02/22/applebug.html